Library/Homebrew/cask/cask.rb

```diff
@@ -131,18 +139,39 @@ def caskfile_only?
 
     sig { returns(T.nilable(Time)) }
     def install_time
-      _, time = timestamped_versions.last
-      return unless time
-
-      Time.strptime(time, Metadata::TIMESTAMP_FORMAT)
+      time = installed_caskfile&.dirname&.dirname&.basename&.to_s
+      Time.strptime(time, Metadata::TIMESTAMP_FORMAT) if time
     end
 
+    sig { returns(T.nilable(Pathname)) }
     def installed_caskfile
-      installed_version = timestamped_versions.last
-      caskfile_dir = metadata_main_container_path.join(*installed_version, "Casks")
-      return caskfile_dir.join("#{token}.json") if caskfile_dir.join("#{token}.json").exist?
+      installed_caskroom_path = caskroom_path
+      installed_token = token
+
+      # Check if the cask is installed with an old name.
+      old_tokens.each do |old_token|
```

Comment:
Should we check for the existence of the default installed caskroom path before entering this loop?

Reply:
It shouldn't normally happen (probably only if migration fails for some reason), but old versions should be preferred in case a directory exists for both old and new names.